Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm

```diff
@@ -149,35 +175,171 @@ fragment float4 fragmentShader(RasterizerData in [[stage_in]],
+            cameraDevice->implData = std::make_unique<CameraDevice::ImplData>();
+            cameraDevice->implData->avDevice = device;
+
+            cameraDevice->capabilities.push_back(std::make_shared<CameraCapabilityTemplate<std::string>>
+            (
+                CameraCapability::Capability::FacingMode,
+                device.position == AVCaptureDevicePositionFront ? "user" : "environment",
+                device.position == AVCaptureDevicePositionFront ? "user" : "environment",
+                device.position == AVCaptureDevicePositionFront ? std::vector<std::string>{"user"} : std::vector<std::string>{"environment"}
+            ));
+            
+            cameraDevice->capabilities.push_back(std::make_shared<CameraCapabilityTemplate<bool>>
+            (
+                CameraCapability::Capability::Torch,
+                false,
+                false,
+                device.isTorchAvailable ? std::vector<bool>{false, true} : std::vector<bool>{false},
+                [this](bool newValue)
+                {
+                    NSError* error{nil};
+                    AVCaptureTorchMode torchMode{ newValue ? AVCaptureTorchModeOn : AVCaptureTorchModeOff };
+                    [m_implData->avCaptureDevice lockForConfiguration:&error];
+                    [m_implData->avCaptureDevice setTorchMode:torchMode];
+                    [m_implData->avCaptureDevice unlockForConfiguration];
+                    if (error != nil)
+                    {
+                        return false;
+                    }
+                    return true;
+                }
+            ));
+            
+#if (TARGET_OS_IPHONE)
+            // iOS Zoom factors always start at 1.0 and go up from there slide the scale based on
+            // the device type (if it starts with an ultrawide sensor or telephoto)
+            double zoomFactorScale{ 1.0 };
+            if (@available(iOS 13.0, *))
+            {
+                if (device.deviceType == AVCaptureDeviceTypeBuiltInTripleCamera ||
+                    device.deviceType == AVCaptureDeviceTypeBuiltInDualWideCamera ||
+                    device.deviceType == AVCaptureDeviceTypeBuiltInUltraWideCamera)
+                {
+                    zoomFactorScale = 0.5;
+                }
+            }
+            if (device.deviceType == AVCaptureDeviceTypeBuiltInTelephotoCamera)
+            {
+                // Newer iOS devices might have a telephoto that has a starting zoom higher, but so far
+                // as I can tell there is no API to determine the default real world zoom of the telephoto lens
+                zoomFactorScale = 2.0;
+            }
+            
+            cameraDevice->capabilities.push_back(std::make_shared<CameraCapabilityTemplate<double>>
+            (
+                CameraCapability::Capability::Zoom,
+                1.0 * zoomFactorScale, // The device always opens at 1.0 which needs to be adjusted for the zoomFactorScale
```

Comment:
Why do this multiplication?  It looks like this can only be 3 values 1.0, 0.5, or 2.0.

Reply:
I'm updating the comment to clarify this.

This line of code is setting the initial zoom factor which when opening the device it will always start at 1.0 according to the scale Apple provides. That scale is dumb though and the value always goes from 1.0+ even if you're using the wide angle lens where 1.0 really means a zoom ratio of 0.5.

I have the zoomFactorScale in place to translate from Apples scale that always starts at 1.0 (regardless of the lens's intrinsic zoom ratio) to our desired scale where the zoom ratio equates to how zoomed in the output image is.